phenol-cli/src/main/java/org/monarchinitiative/phenol/cli/demo/ResnikGenebasedHpoDemo.java

```diff
@@ -0,0 +1,189 @@
+package org.monarchinitiative.phenol.cli.demo;
+
+import com.beust.jcommander.Parameter;
+import com.beust.jcommander.Parameters;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.Multimap;
+import com.google.common.collect.Sets;
+import org.monarchinitiative.phenol.annotations.assoc.HpoAssociationParser;
+import org.monarchinitiative.phenol.annotations.formats.hpo.HpoDisease;
+import org.monarchinitiative.phenol.annotations.obo.hpo.HpoDiseaseAnnotationParser;
+import org.monarchinitiative.phenol.io.OntologyLoader;
+import org.monarchinitiative.phenol.ontology.data.Ontology;
+import org.monarchinitiative.phenol.ontology.data.TermId;
+import org.monarchinitiative.phenol.ontology.data.TermIds;
+import org.monarchinitiative.phenol.ontology.similarity.HpoResnikSimilarity;
+
+import java.io.File;
+import java.util.*;
+
+public class ResnikGenebasedHpoDemo {
+
+  private final Ontology hpo;
+  private final Map<TermId, HpoDisease> diseaseMap;
+  private HpoResnikSimilarity resnikSimilarity;
+  private final Map<TermId, Double> termToIc;
+  private final Multimap<TermId,TermId> geneToDiseaseMap;
+  private final Map<TermId,String> geneIdToSymbolMap;
+  private final Map<TermId, Collection<TermId>> diseaseIdToTermIds;
+
+  public ResnikGenebasedHpoDemo(ResnikGenebasedHpoDemo.Options options) {
+    long start = System.currentTimeMillis();
+    this.hpo = OntologyLoader.loadOntology(new File(options.hpoPath));
+    long now = System.currentTimeMillis();
+    double duration = (double)(now-start)/1000.0;
+    System.out.printf("[INFO] Loaded hp.obo in %.3f seconds.\n",duration);
+    start = now;
+    List<String> databases = ImmutableList.of("OMIM"); // restrict ourselves to OMIM entries
+    this.diseaseMap = HpoDiseaseAnnotationParser.loadDiseaseMap(options.getPhenotypeDotHpoaPath(), hpo,databases);
+    now = System.currentTimeMillis();
+    duration = (double)(now-start)/1000.0;
+    System.out.printf("[INFO] Loaded phenotype.hpoa in %.3f seconds.\n",duration);
+    // Compute list of annoations and mapping from OMIM ID to term IDs.
+    start = now;
+    this.diseaseIdToTermIds = new HashMap<>();
```

Comment:
Consider moving the relevant blocks into individual methods to load each of the required fields, this is a really long constructor and I lost track of that!

Reply:
It is possibly a sign of bad design or just inevitable given the upstream data, but there are a bunch of classes that do more than one thing (parse more than one file to combine them). It is not straightforward to break things up into functions like this without a pretty major rewrite.